lib/mutant/ast/regexp/transformer/quantifier.rb

```diff
@@ -0,0 +1,97 @@
+module Mutant
+  module AST
+    module Regexp
+      class Transformer
+        class Quantifier < self
+          %i[
+            regexp_greedy_zero_or_more
+            regexp_greedy_one_or_more
+            regexp_greedy_zero_or_one
+            regexp_possessive_zero_or_one
+            regexp_reluctant_zero_or_more
+            regexp_reluctant_one_or_more
+            regexp_possessive_zero_or_more
+            regexp_possessive_one_or_more
+            regexp_greedy_interval
+            regexp_reluctant_interval
+            regexp_possessive_interval
+          ].map(&method(:register))
+
+          class ExpressionToAST < Transformer::ExpressionToAST
+            def call
+              ast(expression.min, expression.max)
+            end
+
+          private
+
+            def ast_type
+              :"regexp_#{expression.mode}_#{expression.token}"
+            end
+          end
+
+          class ASTToExpression < Transformer::ASTToExpression
+            include NamedChildren
+
+            children :min, :max, :subject
+
+            QUANTIFIER_MAP = IceNine.deep_freeze(
+              regexp_greedy_zero_or_more:     [:zero_or_more, '*',  :greedy],
+              regexp_greedy_one_or_more:      [:one_or_more,  '+',  :greedy],
+              regexp_greedy_zero_or_one:      [:zero_or_one,  '?',  :greedy],
+              regexp_possessive_zero_or_one:  [:zero_or_one,  '?+', :possessive],
+              regexp_reluctant_zero_or_more:  [:zero_or_more, '*?', :reluctant],
+              regexp_reluctant_one_or_more:   [:one_or_more,  '+?', :reluctant],
+              regexp_possessive_zero_or_more: [:zero_or_more, '*+', :possessive],
+              regexp_possessive_one_or_more:  [:one_or_more,  '++', :possessive],
+              regexp_greedy_interval:         [:interval,     '',   :greedy],
+              regexp_reluctant_interval:      [:interval,     '?',  :reluctant],
+              regexp_possessive_interval:     [:interval,     '+',  :possessive]
+            )
+
+            def call
+              expression = Regexp.to_expression(subject)
```

Comment:
Instance of https://github.com/mbj/mutant/pull/565/commits/b3b6c22e8868baee9691c75127e3baba833ce8c7#r61763338